### PR TITLE
VR Week View - Dynamic multi-day stack support

### PR DESCRIPTION
**Pull request by lucatume**
Screencap: https://drive.google.com/open?id=1-16grcS_RV9rCry16DlxU5D_yv_GovwU&authuser=luca@tri.be&usp=drive_fs

Tickets:
* https://central.tri.be/issues/131100
* https://central.tri.be/issues/131101

This PR updates some code required to support PRO Week View dynamic multi-day stack.

### CODE REVIEW

File: src/Tribe/Views/V2/Views/By_Day_View.php [hunk 1]
```diff
@@ -46,13 +47,23 @@ abstract class By_Day_View extends View{
 	 */
 	protected $grid_days_found_cache = [];
 
+	/**
+	 * An instance of the Stack object.
+	 *
+	 * @since 4.9.7
```
Comment (juanfra) [inline src/Tribe/Views/V2/Views/By_Day_View.php]: Let's use `TBD` here if it's a new addition.

File: src/functions/template-tags/general.php [hunk 2]
```diff
@@ -1097,8 +1098,8 @@ function tribe_events_event_schedule_details( $event = null, $before = '', $afte
 
 			if ( tribe_event_is_all_day( $event ) ) {
 				$inner .= tribe_get_start_date( $event, true, $format );
-				$inner .= '</span>' . $time_range_separator;
-				$inner .= '<span class="tribe-event-date-end">';
+				$inner .= ($html ? '</span>' : '') . $time_range_separator;
```
Comment (juanfra) [inline src/functions/template-tags/general.php]: Let's fix the spacing here
```suggestion
				$inner .= ( $html ? '</span>' : '' ) . $time_range_separator;
```

File: src/functions/template-tags/general.php [hunk 3]
```diff
@@ -1113,8 +1114,8 @@ function tribe_events_event_schedule_details( $event = null, $before = '', $afte
 				$inner .= $end_date;
 			} else {
 				$inner .= tribe_get_start_date( $event, false, $format ) . ( $time ? $datetime_separator . tribe_get_start_date( $event, false, $time_format ) : '' );
-				$inner .= '</span>' . $time_range_separator;
-				$inner .= '<span class="tribe-event-date-end">';
+				$inner .= ($html ? '</span>' : '')  . $time_range_separator;
```
Comment (juanfra) [inline src/functions/template-tags/general.php]: Let's fix the spacing here

```suggestion
				$inner .= ( $html ? '</span>' : '' )  . $time_range_separator;
```

File: src/Tribe/Models/Post_Types/Event.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+				) )->on_resolve( $cache_this ),
+				'plain_schedule_details' => ( new Lazy_String(
+					static function () use ( $post_id )
+					{
+						return tribe_events_event_schedule_details( $post_id, '', '', false );
+					}, false
+				) )->on_resolve( $cache_this )
+			];
+		} catch ( \Exception $e ) {
+			return [];
```
Comment (bordoni) [inline src/Tribe/Models/Post_Types/Event.php]: Should we add logging here?

**Pull request merged**